pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,549 @@
+			Namespace:         "pod1_ns",
+			UID:               "pod4",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+			DeletionTimestamp: &metav1.Time{Time: refTime},
+		},
+	}
+	timestampMap = map[types.UID]*podTimestamps{
+		"pod1": {CreationTimestamp: refTime},
+		"pod2": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+		"pod4": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+	}
+	node = &v1.Node{}
+)
+
+func Test_onPodUpdate(t *testing.T) {
+	newPod1 := &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "4.5.6.7",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "pod1_ns",
+			UID:       "pod1",
+		},
+	}
+	tests := []struct {
+		name        string
+		oldObj      interface{}
+		newObj      interface{}
+		expectedPod *v1.Pod
+	}{
+		{
+			name:        "newObj is not Pod",
+			newObj:      node,
+			expectedPod: pod1,
+		},
+		{
+			name:        "valid case",
+			newObj:      newPod1,
+			expectedPod: newPod1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			podStore := &PodStore{
+				pods: cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			}
+			err := podStore.pods.Add(pod1)
+			require.NoError(t, err)
+			podStore.onPodUpdate(tt.oldObj, tt.newObj)
+			require.Len(t, podStore.pods.List(), 1)
+			assert.Equal(t, tt.expectedPod, podStore.pods.List()[0].(*v1.Pod))
+		})
+	}
+}
+
+func Test_onPodCreate(t *testing.T) {
+	tests := []struct {
+		name        string
+		obj         interface{}
+		expectedMap map[types.UID]*podTimestamps
+	}{
+		{
+			name:        "object is not Pod",
+			obj:         node,
+			expectedMap: map[types.UID]*podTimestamps{},
+		},
+		{
+			name:        "valid case for Pending Pod",
+			obj:         pod1,
+			expectedMap: map[types.UID]*podTimestamps{"pod1": {CreationTimestamp: refTime}},
+		},
+		{
+			name:        "valid case for Running Pod",
+			obj:         pod3,
+			expectedMap: map[types.UID]*podTimestamps{"pod3": {CreationTimestamp: refTime.Add(-5 * time.Minute)}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			podStore := &PodStore{
+				timestampMap: map[types.UID]*podTimestamps{},
+				clock:        clock.NewFakeClock(refTime),
+				pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			}
+			podStore.onPodCreate(tt.obj)
+			assert.Equal(t, tt.expectedMap, podStore.timestampMap)
+		})
+	}
+}
+
+func Test_onPodDelete(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	fakeClock := clock.NewFakeClock(refTime)
+	podStore := NewPodStoreWithClock(podInformer, fakeClock)
+	err := podStore.pods.Add(pod1)
+	require.NoError(t, err)
+	podStore.timestampMap = map[types.UID]*podTimestamps{"pod1": {CreationTimestamp: refTime}}
```

Comment:
duplicate initialization in subtests, it's not good to re-use these objects across tests

Reply:
Fixed.